### PR TITLE
Add MPI call to ArborX

### LINKED ISSUE #60: Bring distributed ArborX search

### PR DESCRIPTION
**Pull request by uhetmaniuk**
Close #60 

- Enable the contact simulation with MPI
- Call ArborX with MPI
- Add GitHub actions for (ArborX, MPI)

### CODE REVIEW

**Comment by uhetmaniuk** [comment]
@rjones-sandia 
I was able to run and "exodiff" the problem sphere_plate_contact for np = 1 and for np = 2.
For np = 2, I re-created the decomposition.
For np = 4 and 8, I still get an error with exodiff. So I will create a separate issue to find a solution.

File: src/contact/parallel/arborx_parallel_contact_manager.cc [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+    //
+    double force[dim] = {0., 0., 0.};
+    if (inside) {
+      ///
+      /// TODO Check the factor 3 with RJ and NM
+      ///
+      details::getContactForce(penalty_ / static_cast<double>(3), gap, normal, force);
+      //
+      if (list_.count(PairData{p_data.rank_, p_data.index_, rank_, f_primitive}) == 0) {
+        list_.insert(PairData{p_data.rank_, p_data.index_, rank_, f_primitive});
```
Comment (dalg24) [inline src/contact/parallel/arborx_parallel_contact_manager.cc:134]:
```suggestion
      auto result = list_.insert(PairData{p_data.rank_, p_data.index_, rank_, f_primitive});
      if (result.second) { // if the insertion took place
```

Comment (uhetmaniuk) [reply src/contact/parallel/arborx_parallel_contact_manager.cc]: Thanks. Done.

**Comment by uhetmaniuk** [comment]
@dalg24 
In the output data, I am carrying the coordinate of the node (or centroid of the bounding box) in double precision.
I am under the impression that ArborX will convert box coordinates to simple precision (float). 
Am I mistaken?

**Comment by dalg24** [comment]
> In the output data, I am carrying the nodal coordinates (or coordinates for the centroid of the bounding box) in double precision.
> I am under the impression that ArborX will convert box coordinates to simple precision (float).
> Am I mistaken?

You are correct.

**Pull request merged**